applications/crossbar/src/modules_v1/cb_users_v1.erl

```diff
@@ -297,6 +341,180 @@ get_channels(Context) ->
+    vcard_escape_chars(Val).
+
+-spec vcard_normalize_type(list() | {ne_binary(), ne_binary()} | ne_binary()) -> ne_binary().
+vcard_normalize_type(T) when is_list(T) -> iolist_join(<<";">>, [vcard_normalize_type(X) || X <- T]);
+vcard_normalize_type({T, V}) -> iolist_join(<<"=">>, [T, V]);
+vcard_normalize_type(T) -> T.
+
+%-spec card_field(ne_binary(), wh_json:object()) -> {vcard_type_spec(), vcard_val()}.
+card_field(Key, _)
+  when Key =:= <<"BEGIN">> ->
+    {Key, <<"VCARD">>};
+card_field(Key, _)
+  when Key =:= <<"VERSION">> ->
+    {Key, <<"3.0">>};
+card_field(Key, _)
+  when Key =:= <<"END">> ->
+    {Key, <<"VCARD">>};
+card_field(Key, JObj)
+  when Key =:= <<"FN">> ->
+    FirstName = wh_json:get_value(<<"first_name">>, JObj),
+    LastName = wh_json:get_value(<<"last_name">>, JObj),
+    MiddleName = wh_json:get_value(<<"middle_name">>, JObj),
+    {Key, iolist_join(<<" ">>, [X || X <- [FirstName, MiddleName, LastName], X =/= undefined, X =/= [], X =/= <<>>])};
+card_field(Key, JObj) when Key =:= <<"N">> ->
+    FirstName = wh_json:get_value(<<"first_name">>, JObj),
+    LastName = wh_json:get_value(<<"last_name">>, JObj),
+    MiddleName = wh_json:get_value(<<"middle_name">>, JObj),
+    {Key, {$;, [LastName, FirstName, MiddleName]}};
+card_field(Key, JObj) when Key =:= <<"ORG">> ->
+    {Key, element(2, card_field(<<"org">>, JObj))};
+card_field(Key, JObj) when Key =:= <<"PHOTO">> ->
+    case card_field(?PHOTO, JObj) of
+        {?PHOTO, 'undefined'} -> {Key, 'undefined'};
+        {?PHOTO, PhotoJObj} ->
+            [{CT, PhotoBin}] = wh_json:to_proplist(PhotoJObj),
+            TypeType = case CT of
+                           <<"image/jpeg">> -> <<"JPEG">>
+                       end,
+            Data = base64:encode(PhotoBin),
+            {[Key, {<<"ENCODING">>, <<"B">>}, {<<"TYPE">>, TypeType}], Data}
+    end;
+card_field(Key, JObj) when Key =:= <<"ADR">> ->
+    Addresses = wh_json:get_value(<<"addresses">>, JObj, []),
+    [{Key, X} || X <- Addresses];
+card_field(Key, JObj) when Key =:= <<"TEL">> ->
+    CallerId = wh_json:get_value(<<"caller_id">>, JObj, wh_json:new()),
+    Internal = wh_json:get_value(<<"internal">>, CallerId),
+    External = wh_json:get_value(<<"external">>, CallerId),
+    [
+     {Key, Internal}
+     ,{Key, External}
+    ];
+card_field(Key = <<"EMAIL">>, JObj) ->
+    {Key, element(2, card_field(<<"email">>, JObj))};
+card_field(Key = <<"BDAY">>, JObj) ->
+    {Key, element(2, card_field(<<"birthday">>, JObj))};
+card_field(Key = <<"NOTE">>, JObj) ->
+    {Key, element(2, card_field(<<"note">>, JObj))};
+card_field(Key = <<"TITLE">>, JObj) ->
+    {Key, element(2, card_field(<<"title">>, JObj))};
+card_field(Key = <<"ROLE">>, JObj) ->
+    {Key, element(2, card_field(<<"role">>, JObj))};
+card_field(Key = <<"TZ">>, JObj) ->
+    {Key, element(2, card_field(<<"timezone">>, JObj))};
+card_field(Key = <<"NICKNAME">>, JObj) ->
+    Val = case element(2, card_field(<<"nicknames">>, JObj)) of
```

Comment:
Or why not just access the property directly? `wh_json:get_value(<<"nicknames">>, JObj, [])`